lsp-mode.el

```diff
@@ -5693,7 +5693,7 @@ EXTRA is a plist of extra parameters.
 REFERENCES? t when METHOD returns references."
   (let ((loc (lsp-request method
                           (append (lsp--text-document-position-params) extra))))
-    (if (seq-empty-p loc)
+    (if (or (null loc) (hash-table-empty-p loc))
```

Comment:
The check of `seq-empty-p` and `(null)` is not equivalent, for example when `loc` is vector.

The problem here is `seq-empty-p` don't know how to deal with hash table, how about define an `cl-defmethod` for `seq-empty-p` with hash table specializer? Or you can define new `lsp-seq-empty?` that works.

Btw, the `loc` here can be either `list`, `vector` or `object` of `plist` or `hash table` depends on if `plist` feature is enabled or not

Reply:
Okay, my elisp is not so great, but I understand if `loc` can be all of these object types.  I'll see if I can follow advice regarding `cl-defmethod` and specializers.  Maybe someone will have a fix before I get to it.

Reply:
See https://github.com/emacs-lsp/lsp-mode/pull/2276#discussion_r508252375